toolkit/tools/pkg/imagecustomizerlib/customizeoverlays_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCustomizeImageOverlays(t *testing.T) {
```

Comment:
You should also add a test that enables SELinux on the image, verifies the equivalency rules are added, and verifies that the upperdir receives the correct label.

Reply:
I have booted the testing image and run:

```
root [ /home/test ]# semanage fcontext -l | grep /var/overlays/etc/upper
/var/overlays/etc/upper = /etc

root [ /home/test ]# semanage fcontext -l | grep /overlays/media/upper
/overlays/media/upper = /media
```

I can see the equivalency rules have been applied properly.

Then I run:

```
root [ /home/test ]# ls -Z /var/overlays/etc/upper
system_u:object_r:auditd_etc_t:s0 audit            system_u:object_r:etc_t:s0 machine-id
       system_u:object_r:etc_t:s0 group            system_u:object_r:etc_t:s0 mtab
       system_u:object_r:etc_t:s0 group-           system_u:object_r:etc_t:s0 nsswitch.conf
    system_u:object_r:shadow_t:s0 gshadow          system_u:object_r:etc_t:s0 pam.d
    system_u:object_r:shadow_t:s0 gshadow-         system_u:object_r:etc_t:s0 ssh
       system_u:object_r:etc_t:s0 issue            system_u:object_r:etc_t:s0 udev
       
------

root [ /home/test ]# ls -Z /etc
              system_u:object_r:etc_t:s0 X11
              system_u:object_r:etc_t:s0 alternatives
              system_u:object_r:etc_t:s0 anacrontab
       system_u:object_r:auditd_etc_t:s0 audit
              system_u:object_r:etc_t:s0 bash.bashrc
              system_u:object_r:etc_t:s0 binfmt.d
             ......
             ......
             .......
              system_u:object_r:etc_t:s0 systemd
              system_u:object_r:etc_t:s0 tdnf
              system_u:object_r:etc_t:s0 terminfo
              system_u:object_r:etc_t:s0 tmpfiles.d
              system_u:object_r:etc_t:s0 udev
          unconfined_u:object_r:etc_t:s0 vimrc
              system_u:object_r:etc_t:s0 xdg
              system_u:object_r:etc_t:s0 yum.repos.d
```

I can see way more labels from `/etc` than its upperdir. Does this make sense? If so, how we can determine the upperdir received the proper labels?

Reply:
Another question is you see these verifications are based on the output of cmd executing, do we have config files that can read and verify, so that I can put the logics into the functional tests?

Reply:
The upperdir will only contain files that have been modified. So, it makes sense that `/etc` will contain more files than the upperdir.

Also, it is expected that a lot of the files/directories under `/etc` have the `etc_t` label and not another more specific label. Specific labels are only created when needed.

Reply:
In terms of the functional tests, it should be sufficient to just verify that the upperdir gets the correct label. And in fact, the upperdir will be empty (since the OS hasn't booted yet). So, there won't be any child files/directories to verify.

Reply:
I have added this function `verifyOverlaysEquivalencyRules`, but saw no dirs error for both overlays:

```
fork/exec ls -Zd /home/lanzeliu/workspace/azurelinux/toolkit/tools/pkg/imagecustomizerlib/_tmp/TestCustomizeImageOverlays/build/testimageroot/var/overlays/etc/upper: no such file or directory
```

This dir exists in my local, I don't understand why here it cannot be found, could you elaborate?

Reply:
Resolved by correcting the sheel.Execute() params.